### PR TITLE
Make sure that NullArray return a single buffer pointer to null.

### PR DESCRIPTION
**Pull request by anliakho2**
Null column must produce n_buffers=1  with a single nullpointer in buffers array. Otherwise `import_array_from_c` would fail with validation error.

### CODE REVIEW

**Comment by jorgecarleitao** [comment]
Thanks.

I wonder who requires a buffer in the null array. E.g. we test this against the C++ implementation and it does not seem to require it: https://github.com/jorgecarleitao/arrow2/blob/main/arrow-pyarrow-integration-testing/tests/test_sql.py#L42

**Comment by anliakho2** [comment]
Also I want to clarify that I'm using a higher level pyarrow api `pyarrow.RecordBatch._import_from_c`

**Comment by jorgecarleitao** [comment]
interesting, because we test exactly this against pyarrow (7.0) [here](https://github.com/jorgecarleitao/arrow2/blob/main/arrow-pyarrow-integration-testing/tests/test_sql.py#L42), but we do not get any error.

Note that I do not disagree with the implementation, I am just trying to find a way to test this

**Comment by anliakho2** [comment]
The test is doing `array._import_from_c` and I'm working on higher level and doing 'RecordBatch._import_from_c'

Ahh found it, it's arrow5, it had hard check for at least one buffer https://github.com/apache/arrow/blob/4591d76fce2846a29dac33bf01e9ba0337b118e9/cpp/src/arrow/c/bridge.cc#L1302
It got removed since arrow6, which now supports single buffer but would also work with 0 buffers

File: src/array/null.rs [hunk 1]
```diff
@@ -109,7 +109,7 @@ impl std::fmt::Debug for NullArray {
 
 unsafe impl ToFfi for NullArray {
     fn buffers(&self) -> Vec<Option<std::ptr::NonNull<u8>>> {
-        vec![]
+        vec![None]
```
Comment (jorgecarleitao) [inline src/array/null.rs:112]:
```suggestion
        // `None` is technically not required by the specification, but older C++ implementations require it, so leaving
        // it here for backward compatibility
        vec![None]
```

**Comment by jorgecarleitao** [comment]
Thanks a lot for sticking and for looking for the root cause, @anliakho2 .

**Comment by anliakho2** [comment]
Pleasure is all mine @jorgecarleitao! And really appreciate quick turnaround and pointers to the test code covering this :)

**Pull request merged**